plugins/inputs/snmp_trap/integration_test/docker-compose.test.yml

```diff
@@ -0,0 +1,40 @@
+#version 3 doesn't have depends_on condition service_healthy
+version: '2.4'
+
+networks:
+  integration:
+    driver: bridge
+    ipam:
+      config:
+        - subnet: 10.0.0.0/8 # test expects netsnmp containter to be 10.0.0.3
+
+services:
+  telegraf_test:
+    build:
+      context: .
+      dockerfile: ./Dockerfile.test
+    networks:
+      - integration
+    volumes:
+      - ../../../..:/go/src/github.com/influxdata/telegraf
+    healthcheck:
+#I haven't been able to get any shell command to work when provided as a string
+#      test: "ss -lun|grep :12399"
+#A list starting with CMD-SHELL is supposed to be equivalent but does work
+      test: ["CMD-SHELL", "ss -lun|grep :12399"]
+      interval: 1s
+      retries: 40
+      #enough to download and build module dependencies,
+      #build and run the test, then open the port
+  netsnmp:
+    build:
+      context: .
+      dockerfile: ./Dockerfile.netsnmp
+    restart: on-failure
+    networks:
+      - integration
+    depends_on:
+      telegraf_test:
+        condition: service_healthy
```

Comment:
You may have already tried this but I saw a [stack overflow](https://stackoverflow.com/questions/47710767/what-is-the-alternative-to-condition-form-of-depends-on-in-docker-compose-versio#:~:text=Version%203%20no%20longer%20supports,that%20behaviour%20using%20V3%20upwards.) which says you can replicate depends on conditional behaviour in v3 using dockerize -wait

Reply:
@helenosheaa dockerize -wait works well for tcp services like http.  It waits for the service to accept connections.  Snmp traps use udp which is connectionless so there's no way to know if telegraf is accepting connections yet.